src/server/plugins/engine/plugin.ts

```diff
@@ -100,155 +117,210 @@ export const plugin = {
-                    forms[key].def.feedback?.feedbackForm
-                  )
-              )
-            )
-          )
-          .code(200)
-      },
-      options: {
-        description: `${enabledString} Gets all published forms. Requires previewMode to be set to true. See runner/README.md for details on environment variables`
-      }
-    })
+    //     const { values } = forms[id]
+    //     return h.response(JSON.stringify({ id, values })).code(200)
+    //   },
+    //   options: {
+    //     description: `${enabledString} Gets a published form, by form id. Requires previewMode to be set to true. See runner/README.md for details on environment variables`
+    //   }
+    // })
+
+    // server.route({
+    //   method: 'get',
+    //   path: '/published',
+    //   handler: (request: Request, h: ResponseToolkit) => {
+    //     if (!previewMode) {
+    //       request.logger.error(
+    //         [`GET /published`, 'previewModeError'],
+    //         disabledRouteDetailString
+    //       )
+    //       throw Boom.unauthorized('publishing is disabled.')
+    //     }
+    //     return h
+    //       .response(
+    //         JSON.stringify(
+    //           Object.keys(forms).map(
+    //             (key) =>
+    //               new FormConfiguration(
+    //                 key,
+    //                 forms[key].name,
+    //                 undefined,
+    //                 forms[key].def.feedback?.feedbackForm
+    //               )
+    //           )
+    //         )
+    //       )
+    //       .code(200)
+    //   },
+    //   options: {
+    //     description: `${enabledString} Gets all published forms. Requires previewMode to be set to true. See runner/README.md for details on environment variables`
+    //   }
+    // })
 
     const queryParamPreHandler = async (
       request: Request,
       h: ResponseToolkit
     ) => {
       const { query } = request
-      const { id } = request.params
-      const model = forms[id]
-      if (!model) {
-        throw Boom.notFound('No form found for id')
-      }
-
+      const model = request.app.model
       const prePopFields = model.fieldsForPrePopulation
-      if (!Object.keys(query).length || !Object.keys(prePopFields).length) {
+      const queryKeysLength = Object.keys(query).length
+      const prePopFieldsKeysLength = Object.keys(prePopFields).length
+
+      if (queryKeysLength === 0 || prePopFieldsKeysLength === 0) {
         return h.continue
       }
+
       const { cacheService } = request.services([])
       const state = await cacheService.getState(request)
       const newValues = getValidStateFromQueryParameters(
         prePopFields,
         query,
         state
       )
+
       await cacheService.mergeState(request, newValues)
+
       return h.continue
     }
 
+    const loadFormPreHandler = server.app.model
+      ? async (request: Request, h: ResponseToolkit) =>
+          (request.app.model = server.app.model)
```

Comment:
not sure if I'm misreading, but this is only caching a single form at a time? What about the other forms?

Reply:
`server.app.model` is only populated in the case where we're exclusively in "run from json file" mode, otherwise we run through the "real" `loadFormPrehandler` method. Inside there we cache models in the `itemsCache` using the key:

```
const key = `${id}_${formState}`
```
E.g. `663cc9a02095c9c7457806ff_draft`, `663cc9a02095c9c7457806ff_live`

[See here](https://github.com/DEFRA/forms-runner/blob/feature/369420-run-forms-from-db/src/server/plugins/engine/plugin.ts#L258)

Reply:
Aha, I understand. I was gettiing `request.app.model` and `server.app.model` mixed up.

Thanks for clarifying.

Reply:
Aha, I understand. I was gettiing request.app.model and server.app.model mixed up.

Thanks for clarifying.